File: Source/Fantasy.Jobs/Resources/TemplateCountResourceProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs.Resources
{
    public class TemplateCountResourceProvider : ConcurrentCountResourceProviderBase
    {
      
        public override bool CanHandle(string name)
        {
            return string.Equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
        }

        public override void Initialize()
        {
            JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
        }

        protected override int GetMaxCount(string key)
        {
            var query = from t in JobManagerSettings.Default.ConcurrentTemplateCount.Templates
                        where key == t.Name
                        select t;
            TemplateCountSetting setting = query.SingleOrDefault();

            return setting != null ? setting.Count : JobManagerSettings.Default.ConcurrentTemplateCount.Count;

        }

        void SettingsPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "ConcurrentTemplateCount")
            {
                this.TryRevoke();
            }

            this.OnAvailable();
        }

        protected override string GetKey(ResourceParameter parameter)
        {
            return parameter.Values["template"];
        }

 
        
    }
}

File: AutoConverted/Fantasy.Jobs/Resources/TemplateCountResourceProvider.java
package Fantasy.Jobs.Resources;

import Fantasy.Jobs.Management.*;

public class TemplateCountResourceProvider extends ConcurrentCountResourceProviderBase
{

	@Override
	public boolean CanHandle(String name)
	{
		return String.equals("RunJob", name, StringComparison.OrdinalIgnoreCase);
	}

	@Override
	public void Initialize()
	{
//C# TO JAVA CONVERTER TODO TASK: Java has no equivalent to C#-style event wireups:
		JobManagerSettings.Default.PropertyChanged += new System.ComponentModel.PropertyChangedEventHandler(SettingsPropertyChanged);
	}

	@Override
	protected int GetMaxCount(String key)
	{
//C# TO JAVA CONVERTER TODO TASK: There is no equivalent to implicit typing in Java:
//C# TO JAVA CONVERTER TODO TASK: There is no Java equivalent to LINQ queries:
		var query = from t in JobManagerSettings.getDefault().getConcurrentTemplateCount().getTemplates() where key.equals(t.getName()) select t;
		TemplateCountSetting setting = query.SingleOrDefault();

		return setting != null ? setting.getCount() : JobManagerSettings.getDefault().getConcurrentTemplateCount().getCount();

	}

	private void SettingsPropertyChanged(Object sender, System.ComponentModel.PropertyChangedEventArgs e)
	{
		if (e.PropertyName.equals("ConcurrentTemplateCount"))
		{
			this.TryRevoke();
		}

		this.OnAvailable();
	}

	@Override
	protected String GetKey(ResourceParameter parameter)
	{
		return parameter.getValues().get("template");
	}



}